src/qibo/tomography/gate_set_tomography.py

```diff
@@ -0,0 +1,236 @@
+                    observable *= obs(q)
+            observable = SymbolicHamiltonian(observable, nqubits=nqubits)
+            expectation_val = result.expectation_from_samples(observable)
+            return expectation_val
+
+
+def execute_GST(
+    nqubits=None,
+    gate=None,
+    nshots=int(1e4),
+    invert_register=None,
+    noise_model=None,
+    backend=None,
+):
+    """Runs gate set tomography for a 1 or 2 qubit gate/circuit.
+
+        Args:
+        nshots (int, optional): Number of shots used in Gate Set Tomography.
+        gate (:class:`qibo.gates.abstract.Gate`, optional): The gate to perform gate set tomography on.
+            If gate=None, then gate set tomography will be performed for an empty circuit.
+        noise_model (:class:`qibo.noise.NoiseModel`, optional): Noise model applied
+            to simulate noisy computation.
+        backend (:class:`qibo.backends.abstract.Backend`, optional): Calculation engine.
+    Returns:
+        ndarray: array with elements ``jk`` equivalent to either :math:`\\text{tr}(Q_{j} \\, \\rho_{k})`
+            or :math:`\\text{tr}(Q_{j} \\, O_{l} \\rho_{k})` where :math:`O_{l}` is the l-th operation
+            in the original circuit.
+    """
+
+    # Check if gate is 1 or 2 qubit gate.
+    if nqubits != 1 and nqubits != 2:
+        raise ValueError(
+            f"nqubits given as {nqubits}. nqubits needs to be either 1 or 2."
+        )
+
+    # Check if invert_register has the correct string.
+    if invert_register is not None:
+        if (
+            invert_register != "sp_0"
+            and invert_register != "sp_1"
+            and invert_register != "sp_t"
+        ):
+            raise NameError(
+                f"{invert_register} not recognized. Input "
+                "sp_0"
+                " to reset qubit 0, "
+                "sp_1"
+                " to reset qubit 1, or "
+                "sp_t"
+                " to reset both qubits."
+            )
+
+    if backend is None:  # pragma: no cover
+        backend = GlobalBackend()
+
+    if gate is not None:
+        qb_gate = len(gate.qubits)
+        if nqubits != qb_gate:
+            raise ValueError(
+                f"Mismatched inputs: nqubits given as {nqubits}. {gate} is a {qb_gate}-qubit gate."
+            )
+        gate = gate.__class__(*list(range(qb_gate)), **gate.init_kwargs)
+
+    # GST for empty circuit or with gates
+    matrix_jk = np.zeros((4**nqubits, 4**nqubits))
+    for k in range(4**nqubits):
+        circ = prepare_states(k, nqubits)
+        if invert_register is not None:
+            inverted_circuit = reset_register(circ, invert_register)
+            if invert_register == "sp_0":
+                circ.add(inverted_circuit.on_qubits(0))
+            elif invert_register == "sp_1":
+                circ.add(inverted_circuit.on_qubits(1))
+            elif invert_register == "sp_t":
+                circ.add(inverted_circuit.on_qubits(0, 1))
```

Comment:
For example if you drop the `sp_X` notation, I believe that this would reduce to:
```suggestion
            circ.add(inverted_circuit.on_qubits(*invert_register))
```

Reply:
But I don't understand, what's the reason of having `sp_0` and `sp_1` when they always correspond to 0 and 1 and, more importantly, they force you to add some `if/else` checks you wouldn't need otherwise?